src/components/playback/playbackmanager.js

```diff
@@ -106,6 +106,17 @@ function normalizeName(t) {
     return t.toLowerCase().replace(' ', '');
 }
 
+function addOrReplaceUrlParam(url, key, value) {
+    // eslint-disable-next-line compat/compat
+    const params = new URLSearchParams((url.split('?')[1] || '').split('#')[0]);
```

Comment:
I think some of this string split and concat could be cleaned up here if we used the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL) and got the search params out of it instead of URLSearchParams directly. The URL object could be used to access the host, path, hash, and search that we're using here.

Reply:
Very belated fix. Note that URL() provides only absolute URLs but that probably isn't an issue in retrospect.